Cython/Compiler/ExprNodes.py

```diff
@@ -2092,7 +2092,7 @@ def check_identifier_kind(self):
         entry = self.entry
         if entry.is_type and entry.type.is_extension_type:
             self.type_entry = entry
-        if entry.is_type and entry.type.is_enum:
+        if entry.is_type and (entry.type.is_enum or entry.type.is_scoped_enum):
```

Comment:
Is a "scoped enum" not an "enum"? I.e. is it better to have separate types and say "or" from time to time, or have scoped enums be a subtype and say "and not" when they should behave differently?

Reply:
I don't feel strongly here. I can make it such that a scoped enum "is a" enum.

Semantically speaking, I guess the correct thing would be to have `is_enum`, `is_scoped_enum`, and `is_unscoped_enum`, where `scoped_enum` and `unscoped_enum` are the two distinct types of `enum`.

Not advocating for this though, just presenting it as a third option if it's appealing to you :)